crates/matrix-sdk-ui/src/room_list_service/mod.rs

```diff
@@ -271,6 +274,26 @@ impl RoomListService {
         self.sliding_sync.stop_sync().map_err(Error::SlidingSync)
     }
 
+    /// Force the sliding sync session to expire.
+    ///
+    /// This is used by [`SyncService`][crate::SyncService].
+    ///
+    /// **Warning**: This method **must** be called when the sync-loop isn't
+    /// running!
```

Comment:
Maybe just me but I find putting the negation earlier makes it clearer.

```suggestion
    /// **Warning**: This method **must not** be called while the sync-loop is
    /// running!
```